app/models/streaming_data_collector.rb

```diff
@@ -25,12 +25,42 @@ def write_conf
   end
 
   def kickoff
-    cmd = "timeout #{Rails.application.config.logstash_run_time} #{Rails.application.config.logstash_command} -f #{filename} &"
-    system(cmd)
+    cmd = "timeout #{Rails.application.config.logstash_run_time} #{Rails.application.config.logstash_command} -f #{filename}"
+
+    # This will detach the process so it doesn't block the main rails process.
+    fork { exec(basic_env, cmd) }
   end
 
   private
 
+  # When we fork, the new process is spawned with the same environment as this
```

Comment:
Really interesting.

Reply:
Glad you like it! It took me an awfully long time to figure out, so hopefully you don't have to spend that long if it ever comes up for you in future :)